src/codegen/compile.ml

```diff
@@ -4903,20 +5027,61 @@ module MakeSerialization (Strm : Stream) = struct
      It returns the value of type t (vanilla representation) or coercion_error_value,
      It advances the data_buffer past the decoded value (even if it returns coercion_error_value!)
    *)
+
+  let with_rel_buf_opt env extended get_typtbl_size1 f =
+    if extended then
+      f (compile_unboxed_const 0l)
+    else
+      get_typtbl_size1 ^^ get_typtbl_size env ^^
+      E.call_import env "rts" "idl_sub_buf_words" ^^
+      Stack.dynamic_with_words env "rel_buf" f
+
+
+  let idl_sub env t2 =
+    let idx = List.length (!(env.E.typtbl_typs)) in
+    env.E.typtbl_typs := !(env.E.typtbl_typs) @ [t2];
+    get_typtbl_idltyps env ^^
+    compile_add_const (Int32.of_int (idx * 4)) ^^
+    G.i (Load {ty = I32Type; align = 0; offset = 0l; sz = None}) ^^
```

Comment:
One could put the offset into `offset` here.

Reply:
Done